### DISCUSSION TITLE
Auto-Import components from `/components` folder

### DISCUSSION THREAD

**Discussion by nathanchase** [withastro/roadmap#258] [Proposal]
In Vue and Svelte, there are packages that do auto-importing of components from the projects's `/components` folder.

Vue has: 
- https://github.com/antfu/unplugin-vue-components
- https://github.com/antfu/unplugin-auto-import
(both of which are [used by Nuxt 3](https://v3.nuxtjs.org/guide/concepts/auto-imports/))

Svelte has:
https://github.com/yuanchuan/sveltekit-autoimport

It would be a great developer experience upgrade. As long as you have no name collisions, all of your components (.astro, .vue, .svelte, etc.) from `/components` would just be available to any Astro component.  Other frameworks have an error check display in the console when such a component naming collision occurs.

Ideally, this would look something like:
```
You have two or more components named 'User':
  '/components/User.astro'
  '/components/User.vue'
Please ensure your component names are unique.
```

**Comment by TechAkayy** [comment]
Unplugin-auto-import is framework agnostic, wouldn't that already work for .astro ato imports?

**Reply by nathanchase** [reply]
It may, but I didn't see any obvious/direct way to integrate it with Astro.

**Reply by delucis** [reply]
This might be of interest: https://www.npmjs.com/package/astro-auto-import

**Reply by nathanchase** [reply]
Yeah, I did see that package, but it seems to only be one component at a time, rather than just using [import.meta.glob()](https://vitejs.dev/guide/migration.html#import-meta-glob) or something to auto-register everything... something like:
```
const componentFiles = import.meta.glob(
  './components/**',
  { eager: true }
);
```

**Reply by nathanchase** [reply]
This article talks about how to do it with Vue+Vite: https://zerotomastery.io/blog/how-to-auto-register-components-for-vue-with-vite/
